core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestRolloverIndexAction.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.rest.action.admin.indices;
+
+import org.elasticsearch.action.admin.indices.rollover.RolloverRequest;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestChannel;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.support.RestToXContentListener;
+
+/**
+ *
+ */
+public class RestRolloverIndexAction extends BaseRestHandler {
+
+    @Inject
+    public RestRolloverIndexAction(Settings settings, RestController controller, Client client) {
+        super(settings, client);
+        controller.registerHandler(RestRequest.Method.POST, "/{index}/_rollover", this);
+        controller.registerHandler(RestRequest.Method.POST, "/{index}/_rollover/{new_index}", this);
```

Comment:
@s1monw applying this patch:
```diff
diff --git a/core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestRolloverIndexAction.java b/core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestRolloverIndexAction.java
index 7eecaaa..77fff1c 100644
--- a/core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestRolloverIndexAction.java
+++ b/core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestRolloverIndexAction.java
@@ -37,14 +37,15 @@ public class RestRolloverIndexAction extends BaseRestHandler {
     @Inject
     public RestRolloverIndexAction(Settings settings, RestController controller, Client client) {
         super(settings, client);
-        controller.registerHandler(RestRequest.Method.POST, "/{index}/_rollover", this);
-        controller.registerHandler(RestRequest.Method.POST, "/{index}/_rollover/{new_index}", this);
+        controller.registerHandler(RestRequest.Method.POST, "/{alias}/_rollover", this);
+        controller.registerHandler(RestRequest.Method.POST, "/{alias}/_rollover/{new_index}", this);
     }
 
     @SuppressWarnings({"unchecked"})
     @Override
     public void handleRequest(final RestRequest request, final RestChannel channel, final Client client) {
-        RolloverRequest rolloverIndexRequest = new RolloverRequest(request.param("index"), request.param("new_index"));
+        logger.info("request params: " + request.params());
+        RolloverRequest rolloverIndexRequest = new RolloverRequest(request.param("alias"), request.param("new_index"));
         if (request.hasContent()) {
             rolloverIndexRequest.source(request.content());
         }
```

I get a rollover request validation exception, when running rest test `indices.rollover/10_basic/Rollover index via API`:
```
{
	"error": {
		"root_cause": [{
			"type": "action_request_validation_exception",
			"reason": "Validation Failed: 1: index alias is missing;"
		}],
		"type": "action_request_validation_exception",
		"reason": "Validation Failed: 1: index alias is missing;"
	},
	"status": 400
}
```
Investigating the elasticsearch log reveals the request params: `{index=logs_search}`, instead of `{alias=logs_search}` causing a request validation exception.